lms/djangoapps/grades/models.py

```diff
@@ -107,7 +107,8 @@ def hash_from_blockrecords(self, blocks):
             blocks = BlockRecordSet(blocks)
 
         try:
-            model = self.create_from_blockrecords(blocks)
+            with transaction.atomic():
+                model = self.create_from_blockrecords(blocks)
```

Comment:
@jcdyer I'd assumed that was from the move to `with transaction.atomic()` in code that does not benefit from the above changes.